x-pack/plugins/apm/public/components/app/service_overview/use_should_use_mobile_layout.ts

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { isWithinMaxBreakpoint } from '@elastic/eui';
+import { useEffect, useState } from 'react';
+
+export function useShouldUseMobileLayout() {
+  const [shouldUseMobileLayout, setShouldUseMobileLayout] = useState(
```

Comment:
fyi UX app has a similar hook. It's in the APM app so easy to import

https://github.com/elastic/kibana/blob/6671cf3c2255c89bc1d2913e62bd06a59d2bd2df/x-pack/plugins/apm/public/components/app/RumDashboard/hooks/useBreakPoints.ts#L12-L35